app/src/main/java/org/simple/clinic/navigation/v2/Router.kt

```diff
@@ -95,9 +95,13 @@ class Router(
       return
     }
 
-    val newHistory = history
-        .removeLast()
-        .add(navRequest)
+    val newHistory = if (history.requests.size > 1) {
+      history
+          .removeLast()
+          .add(navRequest)
+    } else {
+      History(listOf(navRequest))
```

Comment:
Instead of creating a new history, we probably can just return the existing history?

Reply:
When the history contains only one request and we want to replace it with new screen., then we would have to create new history.